<h1>Welcome to the Dark Art of Coding:</h1>
<h2>Introduction to Python</h2>
Session 05

Comma separated value (CSV) files

<img src='../images/logos.3.600.wide.png' height='250' width='300' style="float:right">


<h1>Agenda</h1>

<strong>Review and Questions?</strong>

<strong>Objectives:</strong>

* Open CSVs manually, using straight Python
* Open CSVs using the csv module

<strong>In-class Practice</strong>

# Review and Questions?
---



# Review and Questions?
---

Questions?

* Name three characteristics or capabilities of sets
* Name three tools you can use for "Poor man debugging"
* What command AND what arguments are used to open a file for writing?
* When you are done with a file, how do you put your toys away?
* Name two typical steps (that are often method chained together) to process a csv line, the manual way?

# Manually opening CSVs with straight python

In [1]:
file = open('folder/yahoo_prices_short.csv', 'r')

for line in file:
    fields = line.strip().split(',')
    print(fields[0], fields[5])           
    
# The header row is not quite what we want...    
    

Date Volume
2/29/2008 23860500
2/28/2008 30113200
2/27/2008 27664100
2/26/2008 26013000
2/25/2008 32470600
2/22/2008 26157800
2/21/2008 34494000
2/20/2008 29274700


In [2]:
file = open('folder/yahoo_prices_short.csv')
file.readline()                       # Date,Open,High,Low,Close,Volume,Adj Close

for line in file:
    fields = line.strip().split(',')
    print(fields[0], fields[5])           
    
# this does not help us achieve much in terms of data analysis...

2/29/2008 23860500
2/28/2008 30113200
2/27/2008 27664100
2/26/2008 26013000
2/25/2008 32470600
2/22/2008 26157800
2/21/2008 34494000
2/20/2008 29274700


In [3]:
# A pair of built in functions: max() & min()

l = [1, 2, 3, 4, 42]
print('Max:', max(l))

print('Min:', min(l))


Max: 42
Min: 1


In [4]:
file = open('folder/yahoo_prices_short.csv')
file.readline()

volumes = list()

for line in file:
    fields = line.strip().split(',')
    volumes.append(fields[5])
    
print(max(volumes), min(volumes))    


34494000 23860500


# Opening CSVs with the csv module

## Classic example of what can go wrong!

In [6]:
with open('folder/stocks.csv') as filein:
    filein.readline()           # "Stock Symbol","Stock Name",Open,High,Low,Close,Volume,Adj Close
    open_prices = []            # using list() OR [] is acceptable
    
    for line in filein:
        print(line)
        line = line.strip().split(',')
        print(len(line), line)


APPL,Apple,27.49,28.41,27.5,27.78,23860500,27.78

8 ['APPL', 'Apple', '27.49', '28.41', '27.5', '27.78', '23860500', '27.78']
C,Citigroup,27.98,28.82,27.96,28.15,30113200,28.15

8 ['C', 'Citigroup', '27.98', '28.82', '27.96', '28.15', '30113200', '28.15']
GOOG,Google,30.45,29.81,28.36,30.18,27664100,28.46

8 ['GOOG', 'Google', '30.45', '29.81', '28.36', '30.18', '27664100', '28.46']
HOG,"Harley-Davidson, Inc.",28.48,30.9,27.98,28.48,52354100,28.31

9 ['HOG', '"Harley-Davidson', ' Inc."', '28.48', '30.9', '27.98', '28.48', '52354100', '28.31']
MMM,3m,30.64,27.70,29.11,28.64,82354200,29.74

8 ['MMM', '3m', '30.64', '27.70', '29.11', '28.64', '82354200', '29.74']
M,"Macy's",29.7,26.32,28.16,30.3,72371982,28.12

8 ['M', '"Macy\'s"', '29.7', '26.32', '28.16', '30.3', '72371982', '28.12']
MSFT,Microsoft,26.13,26.75,26.101,25.51,12365478,29.50

8 ['MSFT', 'Microsoft', '26.13', '26.75', '26.101', '25.51', '12365478', '29.50']
WAG,Walgreens,26.63,26.51,28.47,28.33,81271452,26.29
8 ['WAG', 'Walg

<pre>APPL,Apple,27.49,<span style="color:blue">28.41</span>,27.5,27.78,23860500,27.78

C,Citigroup,27.98,<span style="color:blue">28.82</span>,27.96,28.15,30113200,28.15

GOOG,Google,30.45,<span style="color:blue">29.81</span>,28.36,30.18,27664100,28.46

HOG,"Harley-Davidson, Inc.",<span style="color:red">28.48</span>,30.9,27.98,28.48,52354100,28.31

MMM,3m,30.64,<span style="color:blue">27.70</span>,29.11,28.64,82354200,29.74

M,"Macy's",29.7,<span style="color:blue">26.32</span>,28.16,30.3,72371982,28.12

MSFT,Microsoft,26.13,<span style="color:blue">26.75</span>,26.101,25.51,12365478,29.50

WAG,Walgreens,26.63,<span style="color:blue">26.51</span>,28.47,28.33,81271452,26.29</pre>

In [7]:
with open('folder/stocks.csv') as filein:
    filein.readline()
    open_prices = []            # using list() OR [] is acceptable
    
    for line in filein:
        line = line.strip().split(',')
        
        symbol = line[0]
        name = line[1]
        high = line[3]
        
        open_prices.append(float(high))
        
        print(symbol, name, high)

print()        
print('Max:', max(open_prices))
    
# Max 'High' value should have been 30.9 for Harley
# Oooops.

APPL Apple 28.41
C Citigroup 28.82
GOOG Google 29.81
HOG "Harley-Davidson 28.48
MMM 3m 27.70
M "Macy's" 26.32
MSFT Microsoft 26.75
WAG Walgreens 26.51

Max: 29.81


## Let's do it right...

In [8]:
import csv                          # The csv module provides more flexibility and tools than 
                                    # opening the file with straight Python

file = open('folder/stocks.csv')
file.readline()    

csv_stocks = csv.reader(file)       # produces a reader object that parses rows
                                    # the way you expect, out of the box
                                    #     defaults to comma separator
                                    #     understands quoted fields

open_prices = list()

for line in csv_stocks:
    symbol = line[0]
    name = line[1]
    high = line[3]
    
    open_prices.append(high)
    print(symbol, name, high)

    
print('\nMax:', max(open_prices))   # NOTICE the '\n' character

APPL Apple 28.41
C Citigroup 28.82
GOOG Google 29.81
HOG Harley-Davidson, Inc. 30.9
MMM 3m 27.70
M Macy's 26.32
MSFT Microsoft 26.75
WAG Walgreens 26.51

Max: 30.9


In [1]:
import csv                          # The csv module provides more flexibility and tools than 
                                    # opening the file with straight Python

file = open('folder/stocks.tsv')
file.readline()                     # "Stock Symbol","Stock Name",Open,High,Low,Close,Volume,Adj Close

tsv_stocks = csv.reader(file, delimiter='\t', quotechar="'",
                        escapechar="\\")                        
                                                               # csv.reader takes several arguments here
                                                               #     the filehandle
                                                               #     a delimiter character 
                                                               #     a quote character to encapsulate any 
                                                               #     delimiters
                                                               #     an escape character

open_prices = list()

for line in tsv_stocks:
    print(line)
    symbol, name, _open, high, low, close, volume, adjclose = line
    open_prices.append(high)
    print(symbol, name, high)

print('\nMax:', max(open_prices))

['APPL', 'Apple', '27.49', '28.41', '27.5', '27.78', '23860500', '27.78']
APPL Apple 28.41
['C', 'Citigroup', '27.98', '28.82', '27.96', '28.15', '30113200', '28.15']
C Citigroup 28.82
['GOOG', 'Google', '30.45', '29.81', '28.36', '30.18', '27664100', '28.46']
GOOG Google 29.81
['HOG', 'Harley-Davidson, Inc.', '28.48', '30.9', '27.98', '28.48', '52354100', '28.31']
HOG Harley-Davidson, Inc. 30.9
['MMM', '3m', '30.64', '27.70', '29.11', '28.64', '82354200', '29.74']
MMM 3m 27.70
['M', "Macy's", '29.7', '26.32', '28.16', '30.3', '72371982', '28.12']
M Macy's 26.32
['MSFT', 'Microsoft', '26.13', '26.75', '26.101', '25.51', '12365478', '29.50']
MSFT Microsoft 26.75
['WAG', 'Walgreens', '26.63', '26.51', '28.47', '28.33', '81271452', '26.29']
WAG Walgreens 26.51

Max: 30.9


In [ ]:
!less folder/stocks.tsv


'Stock Symbol'  'Stock Name'    Open    High    Low     Close   Volume  Adj Clos e
APPL    Apple   27.49   28.41   27.5    27.78   23860500        27.78
C       Citigroup       27.98   28.82   27.96   28.15   30113200        28.15
GOOG    Google  30.45   29.81   28.36   30.18   27664100        28.46
HOG     'Harley-Davidson, Inc.' 28.48   30.9    27.98   28.48   52354100         28.31
MMM     3m      30.64   27.70   29.11   28.64   82354200        29.74
M       'Macy\'s'       29.7    26.32   28.16   30.3    72371982        28.12
MSFT    Microsoft       26.13   26.75   26.101  25.51   12365478        29.50
WAG     Walgreens       26.63   26.51   28.47   28.33   81271452        26.29
folder/stocks.tsv (END)

In [2]:
# If you're going to read a bunch of CSVs with the same style of formatting then you can
# make a "dialect" which saves some of your arguments

file = open('folder/stocks.tsv')
csv.register_dialect('tsvDialect', delimiter='\t', quotechar="'", escapechar="\\")

tsvinput = csv.reader(file, 'tsvDialect')

for line in tsvinput:
    print(line)

['Stock Symbol', 'Stock Name', 'Open', 'High', 'Low', 'Close', 'Volume', 'Adj Close']
['APPL', 'Apple', '27.49', '28.41', '27.5', '27.78', '23860500', '27.78']
['C', 'Citigroup', '27.98', '28.82', '27.96', '28.15', '30113200', '28.15']
['GOOG', 'Google', '30.45', '29.81', '28.36', '30.18', '27664100', '28.46']
['HOG', 'Harley-Davidson, Inc.', '28.48', '30.9', '27.98', '28.48', '52354100', '28.31']
['MMM', '3m', '30.64', '27.70', '29.11', '28.64', '82354200', '29.74']
['M', "Macy's", '29.7', '26.32', '28.16', '30.3', '72371982', '28.12']
['MSFT', 'Microsoft', '26.13', '26.75', '26.101', '25.51', '12365478', '29.50']
['WAG', 'Walgreens', '26.63', '26.51', '28.47', '28.33', '81271452', '26.29']


In [3]:
import csv                          
csv.register_dialect('tsvDialect', delimiter='\t', quotechar="'", escapechar="\\")

filein = open('folder/stocks.tsv')
fileout = open('folder/stocks_slim.csv', 'w')
filein.readline()                     # "Stock Symbol","Stock Name",Open,High,Low,Close,Volume,Adj Close

tsv_stocks = csv.reader(filein, 'tsvDialect')
stocks_slim = csv.writer(fileout, quotechar='"')

for line in tsv_stocks:
    print(line)
    symbol, name, _open, high, low, close, volume, adjclose = line
    output = [name, symbol, close]
    stocks_slim.writerow(output)

filein.close()
fileout.close()
    

['APPL', 'Apple', '27.49', '28.41', '27.5', '27.78', '23860500', '27.78']
['C', 'Citigroup', '27.98', '28.82', '27.96', '28.15', '30113200', '28.15']
['GOOG', 'Google', '30.45', '29.81', '28.36', '30.18', '27664100', '28.46']
['HOG', 'Harley-Davidson, Inc.', '28.48', '30.9', '27.98', '28.48', '52354100', '28.31']
['MMM', '3m', '30.64', '27.70', '29.11', '28.64', '82354200', '29.74']
['M', "Macy's", '29.7', '26.32', '28.16', '30.3', '72371982', '28.12']
['MSFT', 'Microsoft', '26.13', '26.75', '26.101', '25.51', '12365478', '29.50']
['WAG', 'Walgreens', '26.63', '26.51', '28.47', '28.33', '81271452', '26.29']


# Experience points
---

## Problem 0

1. Create a function that when given an email address can extract the UID (i.e. if given `ballen@jleague.org`, will return `ballen`)
1. Open the file `folder/log_file_1000.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the email address from that row.
1. Print the UID to the screen with the latitude and longitude


In [5]:
import csv

# define function
def getUID(email):    # ballen@jleague
    return email.split("@")[0]   # ['ballen', 'jleague']

# open the file
fin = open('folder/log_file_1000.csv')

# read the csv
mystocks = csv.reader(fin)

# call the function
for fields in mystocks:
    # barry allen,ballen@jleague.org,155.130.121.215,75.122.133.241,2016-02-08T21:44:41,49.8316,8.01485,764272    
    name = fields[0]
    email = fields[1]
    fmip = fields[2]
    uid = getUID(email)
    lat = fields[5]
    long = fields[6]
    print(uid, lat, long)
# print

ballen 49.8316 8.01485
acurry 45.10327 11.68293
jjones 47.11673 10.35874
wwest 46.75616 11.47886
acurry 48.59134 12.30683
rpalmer 45.23082 10.90642
vstone 48.20129 10.54183
krayner 48.8573 9.23887
ckent 48.0599 9.97774
bgordon 46.43865 8.50877
bwayne 49.73236 11.51376
wwest 47.52949 10.77212
rpalmer 47.48082 9.5745
wwest 46.87952 7.83721
ballen 48.03074 10.41633
dprince 47.64781 11.90056
acurry 47.30706 8.47864
jjones 47.89841 11.66265
jjones 46.21873 9.56542
bgordon 49.552 11.59746
dprince 48.60361 9.28674
dprince 45.98007 9.14988
wwest 46.15439 8.44537
dprince 49.89778 8.25535
jjones 47.80911 11.51937
jjones 48.37042 11.56378
hjordan 48.36901 8.26619
hjordan 49.69216 8.72742
kzor-el 46.68394 10.58863
jconstantine 48.83732 12.12413
kzor-el 49.79903 9.05303
jjones 46.75312 12.34015
krayner 49.18048 10.75595
acurry 45.46586 10.67303
hjordan 45.45445 11.0413
jconstantine 46.01856 10.41036
bwayne 45.67383 9.921
hjordan 46.56542 11.1041
vstone 47.16939 9.28906
ckent 49.41465 11.80737
wwest

dprince 46.5187 11.49222
vstone 47.81624 10.25799
dprince 46.02371 11.64555
dprince 45.85027 9.99547
krayner 49.37678 8.92579
jconstantine 46.4143 10.95345
krayner 49.88435 7.93982
dlance 45.26301 7.88211
kzor-el 49.66393 10.29567
dlance 48.41309 11.96555
dlance 45.3885 12.18598
dlance 45.56537 11.65241
dprince 46.30678 11.35342
hjordan 47.72566 8.96686
dgrayson 47.75482 11.42744
skyle 48.10692 9.45693
dlance 46.30996 12.4645
bbatson 47.73574 8.4281
krayner 45.88761 11.86589
dgrayson 46.95896 12.23311
bbatson 47.18423 10.99029
jjones 49.34336 8.83286
vstone 45.50497 11.16468
krayner 48.99076 9.06552
jjones 49.38668 7.83137
kzor-el 49.37511 8.71727
kzor-el 46.88641 9.5728
hjordan 49.67716 7.50511
ckent 45.09485 10.52939
wwest 45.96151 12.05903
acurry 49.17735 10.67473
krayner 46.18192 9.58819
dprince 49.09346 8.93929
dlance 45.40694 11.58039
kzor-el 49.39151 11.2072
ballen 49.89051 11.03083
vstone 49.55589 11.83006
bgordon 46.31914 10.42388
bbatson 45.90089 11.72962
vstone 45.21958 9.54

## Problem 1

1. Create a function that when given an ip address will return True if the ip address is part of the `75.0.0.0/8` network (i.e. check if the first three characters are `"75."`) >>> (see NOTE: below)
1. Open the file `folder/log_file_1000.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the `From IP` address from that row. Also call your function and give it the `To IP` address from that row.
1. Print the name, From IP and To IP and date for only rows where an IP falls into the 75.0.0.0/8 network.


* NOTE: there are way better ways to process IP addresses, this is greatly simplified. see the `ipaddress module`


In [11]:
import csv

def ip75(ipaddress):
    return ipaddress.startswith('75.')
    # ipaddress.split('.')[0]
    # ipaddress[:3] == '75.'

fin = open('folder/log_file_1000.csv')

logs = csv.reader(fin)

for fields in logs:
# barry allen,ballen@jleague.org,155.130.121.215,75.122.133.241,2016-02-08T21:44:41,49.8316,8.01485,764272    
    name = fields[0]
    fmip = fields[2]
    toip = fields[3]
    if ip75(fmip) and ip75(toip):
        print(name, fmip, toip)
    
    

dick grayson 75.122.133.10 75.122.133.75
hal jordan 75.122.133.241 75.122.133.75
arthur curry 75.122.132.124 75.122.133.241
bruce wayne 75.122.133.241 75.122.133.10
dinah lance 75.122.133.241 75.122.133.75
kyle rayner 75.122.133.241 75.122.132.124
arthur curry 75.122.133.241 75.122.133.10
dinah lance 75.122.133.75 75.122.132.124
barry allen 75.122.133.10 75.122.132.124
john constantine 75.122.133.10 75.122.133.10
kara zor-el 75.122.132.124 75.122.133.10
victor stone 75.122.133.75 75.122.133.75
victor stone 75.122.132.124 75.122.133.10
kyle rayner 75.122.133.10 75.122.133.10
john jones 75.122.133.75 75.122.132.124


In [7]:
ipaddr = '75.0.0.1'
ipaddr.startswith('76.')

False

## Problem 2

1. Create a function that when given two text values (lat and long) returns two values: 
   1. Convert each text value to a float
   1. Use the round() function to round the float to two decimal places
1. Open the file `folder/log_file_1000.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the lat and long values
1. Print the date and the lat and long.


## Problem 3

1. Create a function that when given two text values (high stock price and low stock price) returns one value: 
   1. Convert each text value to a float
   1. Subtract the low value from the high value and return the delta
1. Open the file `folder/yahoo_price_short.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the high and low values (NOTE: the fields are ordered: date, open, high, low, close, volume, adj_close)
1. Print the date and the delta between the high and low.




## Problem 4

1. Create a function that when given one text value (volume) returns one value: 
   1. Convert each text value to an integer
   1. Divide the value by 1,000,000
   1. Use the round() function to round to one decimal place
   1. Convert the float to a string and concatenate with an `M` to convert the value to a human readable form (i.e. 28345623 becomes 28.3M)
   1. Subtract the low value from the high value and return the delta
1. Open the file `folder/yahoo_price_short.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the volume (NOTE: the fields are ordered: date, open, high, low, close, volume, adj_close)
1. Print the date and volume in human readable form.





## Problem 5

1. Create a function that when given one text value (date) returns one value: 
   1. Split the text and extract the day
   1. return the day of the month (i.e. 2/29/2008 >>> 29)
1. Open the file `folder/yahoo_price_short.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the date (NOTE: the fields are ordered: date, open, high, low, close, volume, adj_close)
1. Track the highest volume and track which day of the month it fell on.
1. Print the day of the month with the highest volume AND the highest volume

# Solutions:
---

## Problem 0

1. Create a function that when given an email address can extract the UID (i.e. if given `ballen@jleague.org`, will return `ballen`)
1. Open the file `folder/log_file_1000.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the email address from that row.
1. Print the UID to the screen with the latitude and longitude


In [ ]:
import csv

# def getUID(txt):
#     fields = txt.split('@')
#     UID = fields[0]
#     return UID

def getUID(txt):
    return txt.split('@')[0]

with open("folder/log_file_1000.csv") as fin:
    logs = csv.reader(fin)
    
    for line in logs:
        name, email, fmip, toip, date, lat, long, payload = line
        uid = getUID(email)
        print(uid, lat, long)
    

## Problem 1

1. Create a function that when given an ip address will return True if the ip address is part of the `75.0.0.0/8` network (i.e. check if the first three characters are `"75."`) >>> (see NOTE: below)
1. Open the file `folder/log_file_1000.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the `From IP` address from that row. Also call your function and give it the `To IP` address from that row.
1. Print the name, From IP and To IP and date for only rows where an IP falls into the 75.0.0.0/8 network.


* NOTE: there are way better ways to process IP addresses, this is greatly simplified. see the `ipaddress module`


In [ ]:
import csv

# def ip75(txt):
#    if txt.startswith('75.') == True:
#        return True

def ip75(txt):
    return txt.startswith('75.')

with open("folder/log_file_1000.csv") as fin:
    logs = csv.reader(fin)
    
    for line in logs:
        name, email, fmip, toip, date, lat, long, payload = line
        # innetwork1 = ip75(fmip)
        # innetwork2 = ip75(toip)
        # if innetwork1 or innetwork2:
        #     print(name, fmip, toip)

        if ip75(fmip) or ip75(toip):
            print(name, fmip, toip)    
            



## Problem 2

1. Create a function that when given two text values (lat and long) returns two values: 
   1. Convert each text value to a float
   1. Use the round() function to round the float to two decimal places
1. Open the file `folder/log_file_1000.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the lat and long values
1. Print the date and the lat and long.


In [ ]:
import csv

def geoConverter(lat, long):
    lat = round(float(lat), 2)
    long = round(float(long), 2)
    return lat, long
    

with open("folder/log_file_1000.csv") as fin:
    logs = csv.reader(fin)
    
    for line in logs:
        name, email, fmip, toip, date, lat, long, payload = line
        lat, long = geoConverter(lat, long)
        print(date, lat, long)


## Problem 3

1. Create a function that when given two text values (high stock price and low stock price) returns one value: 
   1. Convert each text value to a float
   1. Subtract the low value from the high value and return the delta rounded to 1 decimals
1. Open the file `folder/yahoo_price_short.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the high and low values (NOTE: the fields are ordered: date, open, high, low, close, volume, adj_close)
1. Print the date and the delta between the high and low.




In [ ]:
import csv

def delta(high, low):
    return round(float(high) - float(low), 1)
    
with open("folder/yahoo_prices_short.csv") as fin:
    fin.readline()
    logs = csv.reader(fin)
    
    for line in logs:
        date, _open, high, low, close, volume, adj_close = line
        print(date, delta(high, low))

## Problem 4

1. Create a function that when given one text value (volume) returns one value: 
   1. Convert each text value to an integer
   1. Divide the value by 1,000,000
   1. Use the round() function to round to one decimal place
   1. Convert the float to a string and concatenate with an `M` to convert the value to a human readable form (i.e. 28345623 becomes 28.3M)
   1. Subtract the low value from the high value and return the delta
1. Open the file `folder/yahoo_price_short.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the volume (NOTE: the fields are ordered: date, open, high, low, close, volume, adj_close)
1. Print the date and volume in human readable form.





In [ ]:


import csv

def human(txt):
    num = int(txt)/1000000
    num = round(num, 1)
    return str(num) + 'M'
    
with open("folder/yahoo_prices_short.csv") as fin:
    fin.readline()
    logs = csv.reader(fin)
    
    for line in logs:
        date, _open, high, low, close, volume, adj_close = line
        print(date, human(volume))

## Problem 5

1. Create a function that when given one text value (date) returns one value: 
   1. Split the text and extract the day
   1. return the day of the month (i.e. 2/29/2008 >>> 29)
1. Open the file `folder/yahoo_price_short.csv` AND label it with a suitable filehandle
1. Use the `csv` module to read the data from the file (don't forget to import the module)
1. For each row, call your function AND give it the date (NOTE: the fields are ordered: date, open, high, low, close, volume, adj_close)
1. Track the highest volume and track which day of the month it fell on.
1. Print the day of the month with the highest volume AND the highest volume

In [ ]:
import csv

def dayFromDate(date):
    return date.split('/')[1]
    
with open("folder/yahoo_prices_short.csv") as fin:
    fin.readline()
    logs = csv.reader(fin)
    
    highest_vol = 0
    
    for line in logs:
        date, _open, high, low, close, volume, adj_close = line
        volume = int(volume)
        if volume > highest_vol:
            highest_vol = volume
            day = dayFromDate(date)
        
        
    print(day, highest_vol)
